src/Till.js

```diff
@@ -0,0 +1,50 @@
+var Till = function() {
+  this.detailsList = hipsterCoffeeDetails;
+  this.orderedItems = [];
+  this.subTotal = 0;
+  this.totalCalculator = new TotalCalculator(this.detailsList, this.orderedItems, this.subTotal);
+};
+
+Till.prototype.retrievePrice = function(item) {
+  return this.detailsList[0].prices[0][item];
+};
+
+Till.prototype.addItem = function(item) {
+  this.subTotal += this.retrievePrice(item);
+  this.orderedItems.push(item);
+  return item;
+};
+
+Till.prototype.calculateSubTotal = function() {
+  return this.subTotal;
```

Comment:
When you make the totalCalculator, orderedItems will always be empty - so, make the totalCalculator at the point that it needs to be used.